src/pydap/handlers/dap.py

```diff
@@ -105,13 +107,22 @@ def determine_protocol(self, protocol):
         elif self.scheme == "dap4":
             self.scheme = "http"
             return "dap4"
+        elif self.query[:4] == "dap4":
+            self.sheme = "http"
+            return "dap4"
         else:
             extension = self.path.split(".")[-1]
             if extension in ["dmr", "dap"]:
                 return "dap4"
             elif extension in ["dds", "dods"]:
                 return "dap2"
             else:
+                _warnings.warn(
```

Comment:
This is in case protocol is None. Then pydap picks DAP2 (because at this point more of DAP2 is functional within pydap. But I figure that this warning should inform people about what they are using. Later on, we should make DAP4 the default.